src/components/select-field/BaseSelectField.js

```diff
@@ -247,22 +281,25 @@ class BaseSelectField extends React.Component<Props, State> {
                 }
                 break;
             default: {
-                stopDefaultEvent(event);
-                const lowerCaseKey = key.toLowerCase();
-                const optionIndex = options.findIndex(
-                    option => option.displayText.toLowerCase().indexOf(lowerCaseKey) === 0,
-                );
+                if (!shouldShowSearchInput) {
```

Comment:
If the user has toggled shouldShowSearchInput to be on, this means that we want to allow character keypresses to go through without performing the default setActiveItem behavior. This is because we want the search input field to be able to keep track of what the current search string is.

If the toggle is off, then we still proceed how we did before.